src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,3 +389,124 @@ def get_policy_assignment_completion_list(prefix, **kwargs):#pylint: disable=unu
     result = policy_client.policy_assignments.list()
     return [i.name for i in result]
 
+class _ResourceUtils(object): #pylint: disable=too-many-instance-attributes
+    def __init__(self, resource_group_name=None, resource_provider_namespace=None,
```

Comment:
Don't we validate the provided --id is in the correct format (/subscriptions/subId/resourceGroups/rgName/providers/Microsoft.Compute/vm) before making the request?